### PR TITLE
Weather App Clone 

### PR DESCRIPTION
**Pull request by Arunlalpp**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import './App.css';
-import Navbar from './components/Navbar';
+import "./App.css";
```
Comment (vonnue-sijin) [inline src/App.js:1]: Fetch farthest resource first.

File: src/components/CardContainer.js [hunk 2]
```diff
@@ -69,13 +71,117 @@ export default function CardContainer() {
...
       </div>
+      <div className="bg-white w-9xl m-auto rounded-lg md:hidden block">
+        <h2 className="text-black text-xl font-bold pb-4">Top Stories</h2>
+        <div className="md:grid grid-cols-4 md:gap-12 flex flex-row gap-2 overflow-auto overflow-y-hidden scrollbar-hide scroll-smooth">
+          <div class="rounded-lg  max-w-2xl p-2">
+            <div className="md:min-w-[0] min-w-[72vw]">
+              <div>
+                <img
+                  className="md:h-[15vh] w-[100%] rounded-lg"
+                  src="https://s.w-x.co/in-kerala%20rain.jpg?crop=16:9&width=320&format=pjpg&auto=webp&quality=60"
```
Comment (vonnue-sijin) [inline src/components/CardContainer.js]: Alt tag?

File: src/components/CardContainer.js [hunk 3]
```diff
@@ -69,13 +71,117 @@ export default function CardContainer() {
...
             <img src="https://s.w-x.co/promo-modules-units-features.png?crop=2.85:1&width=320&format=pjpg&auto=webp&quality=60" />
           </div>
         </div>
       </div>
+      <div className="bg-white w-9xl m-auto rounded-lg md:hidden block">
+        <h2 className="text-black text-xl font-bold pb-4">Top Stories</h2>
+        <div className="md:grid grid-cols-4 md:gap-12 flex flex-row gap-2 overflow-auto overflow-y-hidden scrollbar-hide scroll-smooth">
+          <div class="rounded-lg  max-w-2xl p-2">
+            <div className="md:min-w-[0] min-w-[72vw]">
+              <div>
```
Comment (vonnue-sijin) [inline src/components/CardContainer.js]: Purpose of this div?

File: src/components/CardContainer.js [hunk 4]
```diff
@@ -69,13 +71,117 @@ export default function CardContainer() {
...
+              </div>
+              <div className="mt-3 hover:text-blue-500 cursor-pointer md:text-[14px] text-[13px]  ">
+                <p>
+                  Monsoon Spike: Heavy Rains to Lash Kerala and Karnataka Until
+                  June 11; Tamil Nadu, Andhra In for Scattered
+                </p>
+              </div>
+            </div>
+          </div>
+        </div>
```
Comment (vonnue-sijin) [inline src/components/CardContainer.js]: Could make this into a component and reuse,

File: src/components/CardTitle.js [hunk 5]
```diff
@@ -1,61 +1,76 @@
 import React from "react";
+import data from "../data";
+import { useState } from "react";
```
Comment (vonnue-sijin) [inline src/components/CardTitle.js]: Can be imported along with first line itself.

File: src/components/CardTitle.js [hunk 6]
```diff
@@ -1,61 +1,76 @@
...
-                <div className="">
-                  <p>
-                    Heatwaves Return! To Grip Parts of Maharashtra, Madhya
-                    Pradesh, Uttar Pradesh, Punjab, Rajasthan, Haryana from June
-                    3-5
-                  </p>
-                </div>
-              </div>
-              <div className="">
+  const [showMoreButton, setShowMoreButton] = useState(data.data.length > 4);
```
Comment (vonnue-sijin) [inline src/components/CardTitle.js]: No magic numbers store number 4 in a variable. Make code readable.

File: src/components/CardTitle.js [hunk 7]
```diff
@@ -1,61 +1,76 @@
...
+  const loadMoreData = (
+    <>
+      <div className="grid lg:grid-cols-4 gap-12 mt-8">
+        {data.data.map((item, i) => {
+          return (
+            i > 4 && (
+              <>
                 <div>
-                  <img className="rounded-lg" src="https://s.w-x.co/in-heat_14.jpg?crop=16:9&width=640&format=pjpg&auto=webp&quality=60" />
+                  <div>
```
Comment (vonnue-sijin) [inline src/components/CardTitle.js]: Empty divs

File: src/components/CardTitle.js [hunk 8]
```diff
@@ -1,61 +1,76 @@
...
+  const [showMoreButton, setShowMoreButton] = useState(data.data.length > 4);
+
+  const loadMoreData = (
+    <>
+      <div className="grid lg:grid-cols-4 gap-12 mt-8">
+        {data.data.map((item, i) => {
+          return (
+            i > 4 && (
+              <>
                 <div>
```
Comment (vonnue-sijin) [inline src/components/CardTitle.js]: Add Key attribute

File: src/components/CardTitle.js [hunk 9]
```diff
@@ -1,61 +1,76 @@
...
-                    Heatwaves Return! To Grip Parts of Maharashtra, Madhya
-                    Pradesh, Uttar Pradesh, Punjab, Rajasthan, Haryana from June
-                    3-5
-                  </p>
+              </>
+            )
+          );
+        })}
+      </div>
+      {!showMoreButton && (
```
Comment (vonnue-sijin) [inline src/components/CardTitle.js]: Follow boolean naming conventions. Like shouldShowMoreButton

File: src/components/CardTitle.js [hunk 10]
```diff
@@ -1,61 +1,76 @@
...
+    </>
+  );
+
+  return (
+    <div className="bg-white lg:w-[52%] lg:ml-[12.5%] w-9xl ml-[3.3%] rounded-b-lg">
+      <div className="p-4 ">
+        <div className="md:grid grid-cols-4 md:gap-12 flex flex-row gap-2 overflow-auto overflow-y-hidden scrollbar-hide scroll-smooth">
+          {data.data.map((item, i) => {
+            return (
+              i < 4 && (
```
Comment (vonnue-sijin) [inline src/components/CardTitle.js]: Map is not the correct function to be used here. Use a while or something and assign it into an variable and use that here.

File: src/components/CardTitle.js [hunk 11]
```diff
@@ -1,61 +1,76 @@
...
+                <div  className="md:min-w-[0] min-w-[40vw]">
+                  <div>
+                    <img className="rounded-lg md:h-[15vh] w-[100%]" src={item.image} />
+                  </div>
+                  <div className="mt-3 hover:text-blue-500 cursor-pointer md:text-[14px] text-[13px]  ">
+                    <p>{item.title}</p>
```
Comment (vonnue-sijin) [inline src/components/CardTitle.js]: Refrain from using generic names like 'item'.

File: src/components/FooterSvg.js [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+              xmlns="http://www.w3.org/2000/svg"
+              viewBox="0 0 24 24"
+            >
+              <title>Youtube</title>
+              <path
+                fill="currentColor"
+                d="M12.35 5c1.34.008 6.125.06 7.462.42.86.231 1.538.913 1.768 1.78.39 1.465.416 4.415.418 4.802v.085c-.002.387-.028 3.337-.418 4.803a2.513 2.513 0 01-1.768 1.78c-1.374.37-6.391.414-7.566.42h-.495c-1.175-.006-6.192-.05-7.566-.42a2.513 2.513 0 01-1.769-1.78c-.376-1.413-.413-4.208-.417-4.752v-.187c.004-.545.041-3.339.417-4.752a2.513 2.513 0 011.769-1.78c1.336-.36 6.122-.411 7.462-.419zM9.952 9.07v5.948l5.227-2.973L9.953 9.07z"
+              ></path>
+            </svg>
+          </a>
```
Comment (vonnue-sijin) [inline src/components/FooterSvg.js:92]: I think this section can be a component.

File: src/components/IndianMap.js [hunk 13]
```diff
@@ -1,25 +1,34 @@
...
-              </p>
-            </div>
+      <div className="lg:w-[52%] lg:ml-[12.5%] w-9xl ml-[3.3%]">
+        <div className="p-4 rounded-t-lg bg-white">
+          <span className="text-black text-xl font-bold">
+            Weather Across India
+          </span>
+          <img
+            src="https://s.w-x.co/in-saturday.jpg?crop=16:9&width=980&format=pjpg&auto=webp&quality=60"
+            alt="indian-map"
```
Comment (vonnue-sijin) [inline src/components/IndianMap.js]: alt text should be human understandable text.

File: src/components/SmallCard.js [hunk 14]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import Carddata from "../Carddata";
+
+export default function SmallCard() {
+  return (
+    <>
+      <div className="bg-white rounded-lg mt-2 lg:w-[52%] lg:ml-[12.5%] w-9xl ml-[3.3%]">
+        <span className="font-bold m-5">Don't Miss</span>
+        <div className="md:grid grid-cols-4 md:gap-4 md:p-5 p-3 flex flex-row gap-2 overflow-auto overflow-y-hidden scrollbar-hide scroll-smooth">
+          {Carddata.cardData.map((item) => {
```
Comment (vonnue-sijin) [inline src/components/SmallCard.js]: variable to have camelCasing. cardData

File: src/components/VideoCardContainer.js [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+import {ImPlay2} from 'react-icons/im'
+
+export default function VideoCardContainer({ title }) {
+  return (
+    <>
+      <div className="bg-white rounded-lg mt-2 lg:w-[52%] lg:ml-[12.5%] w-9xl ml-[3.3%]">
+        <span className="font-bold ml-5 mt-4">{title}</span>
+        <div className="md:grid md:grid-cols-4 gap-4 p-5 hidden ">
+          <div>
+            <div>
```
Comment (vonnue-sijin) [inline src/components/VideoCardContainer.js]: Empty divs

File: src/components/VideoCardContainer.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+              <ImPlay2 className="absolute top-[40%] left-[40%] text-[4rem] fill-white" />
+            </div>
+            <div className="mt-2">
+              <p className="md:text-[14px] text-[16px] hover:text-blue-500 cursor-pointer">This Plant in Australia Covers An Area Four Times the Size of New Delhi!</p>
+            </div>
+          </div>
+      </div>
+      <button className="md:hidden block ml-6 mb-4 mt-2 rounded-full px-6 py-2.5 bg-[#1B4DE4] text-white font-medium text-xs leading-tight  hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out">Show more</button>
+      </div>
+        {/* <button className="md:block  ml-6 mb-4 inline-block mt-6 rounded-full px-6 py-2.5 bg-[#1B4DE4] text-white font-medium text-xs leading-tight  hover:shadow-lg focus:bg-blue-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-blue-800 active:shadow-lg transition duration-150 ease-in-out">Show more</button> */}
```
Comment (vonnue-sijin) [inline src/components/VideoCardContainer.js]: Remove comments

File: src/components/VideoCardContainer.js (hunk 15 above)
Comment (Arunlalpp) [reply src/components/VideoCardContainer.js]: Need one div for the mobile view this container card will goes one card, so need this div

File: src/components/FooterSvg.js (hunk 12 above)
Comment (Arunlalpp) [reply src/components/FooterSvg.js:92]: Its a  component used in the footer section for reducing the SVG files multiple times use

File: src/components/IndianMap.js (hunk 13 above)
Comment (Arunlalpp) [reply src/components/IndianMap.js]: understood

File: src/components/CardTitle.js (hunk 6 above)
Comment (Arunlalpp) [reply src/components/CardTitle.js]: Thanks for the comment!

File: src/components/CardContainer.js (hunk 4 above)
Comment (Arunlalpp) [reply src/components/CardContainer.js]: Thanks for the idea

**Pull request merged**